pkg/intoto/metadata.go

```diff
@@ -0,0 +1,97 @@
+package intoto
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	canonicaljson "github.com/docker/go/canonical/json"
+)
+
+// Metadata represents the In-Toto metadata stored in TUF.
+// All fields are represented as []byte in order to be stored in the Custom field for TUF metadata.
+type Metadata struct {
+	// TODO: remove this once the TUF targets key is used to sign the root layout
+	Key    []byte            `json:"key"`
+	Layout []byte            `json:"layout"`
+	Links  map[string][]byte `json:"links"`
+}
+
+// WriteMetadataFiles writes the content of a metadata object into files in a directory
+func WriteMetadataFiles(m *Metadata, dir string) error {
+	abs, err := filepath.Abs(dir)
+	if err != nil {
+		return err
+	}
+
+	//FIXME: no need to actually write filename.
```

Comment:
I get that now the verification process takes `*.layout`, but under what name _should_ we write this file?
Same for below.